signer/signer_test.go

```diff
@@ -1 +1,89 @@
 package signer
+
+import (
+	"crypto/x509/pkix"
+	"reflect"
+	"testing"
+
+	"github.com/cloudflare/cfssl/csr"
+)
+
+func TestReplaceSliceIfEmpty(t *testing.T) {
+	contents := &[]string{"foo", "bar"}
+	empty := &[]string{}
+	nonEmpty := &[]string{"foobar"}
+
+	replaceSliceIfEmpty(empty, contents)
+
+	if !reflect.DeepEqual(*empty, *contents) {
+		t.Fatal("Failed to replace empty slice")
+	}
+
+	replaceSliceIfEmpty(nonEmpty, contents)
+
+	if reflect.DeepEqual(*nonEmpty, *contents) {
+		t.Fatal("Replaced non-empty slice")
+	}
+
+}
+
+func TestPopulateSubjectFromCSR(t *testing.T) {
```

Comment:
And consider duplicating this test logic at signer/local/local_test.go TestOverrideSubject(). Let's use tests to specify the expected behavior of signers, otherwise bugs may be introduced later without being noticed.